retrieval/discovery/consul.go

```diff
@@ -191,7 +184,7 @@ func (cd *ConsulDiscovery) watchServices(update chan<- *consulService) {
 		})
 		if err != nil {
 			log.Errorf("Error refreshing service list: %s", err)
-			<-time.After(consulRetryInterval)
+			time.Sleep(consulRetryInterval)
```

Comment:
Let's not pile even more on this PR. Same for the file stuff down there.
The whole retrieval package needs another pass so we will get to it later.